### PR TITLE
Fix `uninlined_format_args` in compiler crates with the diagnostic migration completed

### PR DESCRIPTION
**Pull request by Noratrieb**
Convert all the crates that have had their diagnostic migration completed (except save_analysis because that will be deleted soon and apfloat because of the licensing problem).

Some of them have been reviewed by myself and they were all correct (though I still recommend going over all of them again for review).

### CODE REVIEW

File: compiler/rustc_codegen_llvm/src/llvm_util.rs [hunk 1]
```diff
@@ -327,11 +327,11 @@ fn print_target_features(sess: &Session, tm: &llvm::TargetMachine) {
 
     println!("Features supported by rustc for this target:");
     for (feature, desc) in &rustc_target_features {
-        println!("    {1:0$} - {2}.", max_feature_len, feature, desc);
+        println!("    {feature:max_feature_len$} - {desc}.");
```
Comment (Nilstrieb) [inline compiler/rustc_codegen_llvm/src/llvm_util.rs:330]: That's a fun one, you rarely see it. Looks correct though.

**Comment by Nilstrieb** [comment]
I looked through all the changes and they look fine, as expected.

One problem with this is that this will probably cause a bunch of conflicts with diagnostic migrations which is pretty bad. I'm not sure what the best way to proceed is here. Maybe it makes sense to only do the change on crates that have their diagnostics fully migrated?

**Comment by Nilstrieb** [comment]
As a user for rust-analyzer, the problems aren't bad enough to not do this now in my opinion, but I could also understand it if we deem it bad enough and want to delay it.

**Comment by RalfJung** [comment]
I don't understand why the bot pinged me, I see no interpreter changes?

**Comment by Nilstrieb** [comment]
I had some at first that I then removed.

**Comment by jackh726** [comment]
r=me after rebase 

@bors delegate+

**Comment by Nilstrieb** [comment]
@bors r=jackh726

Btw, I have bors permissions already, no need to delegate :D

**Pull request merged**